### ISSUE TITLE
Signature not being added to the Staff reply

### ISSUE THREAD

**Issue by esamo** [osTicket/osTicket#4097]
Hi

Staff were complaining that their signature is not added to the reply.
And it is not being added indeed. Signature goes to the mail sent, but not to the reply itself.

I've fixed that by adding `$signature` to the `$response` in `Ticket` class -> `postReply` function + moving everything that is needed to prepare signature (`$dept` var declaration).

http://take.ms/G4Um9

---

Config & versions http://take.ms/xmqOJ

**Comment by esamo** [comment]
But then the signature shows twice in the mail :).

**Comment by ntozier** [comment]
ummmm... what version are you running?

Because:
![4097](https://user-images.githubusercontent.com/1315768/33234727-7a74ad22-d1f9-11e7-845f-ab104e07896c.png)

**Comment by esamo** [comment]
Yes it shows nicely in this window but does not get added to the reply. I've pasted a screenshot of the System Information page in the original post.

**Comment by JediKev** [comment]
@esamo are you viewing the email in the Gmail (either the browser or phone app)?

**Comment by esamo** [comment]
@JediKev My problem is not the email but the reply itself in the system, as shown on the screen below. I'm sorry if I was unclear.
The email is OK, but in the system signature doesn't get added to the message. It was so back in version 1.6, therefore after the update Staff got confused and reported that as an error :).

http://take.ms/Jsl5v

**Comment by JediKev** [comment]
Ohhhhh. Yea we don't add the signature there. It just adds more clutter and length to the thread. If you would like the signature there you can make a mod to add it. I don't think this will be accepted as a feature request either because like I said it just adds more length and clutter to the thread. I hope this helps! Cheers.

**Comment by Lineflyer** [comment]
Hi everyone,

I am using osticket V1.12.3 and have a similar but different problem:
- Created a signature for an agent 
- While replying to a ticket the agent can select the signature and it is shown in the editor
- The signature is not sent out with the mail
- Happens with both user and department signature

Any idea or any more information needed?

**Comment by vladh** [comment]
I have the same problem as @Lineflyer. Kind of irritating as I only realised this after sending out hundreds of support emails.

**Comment by JediKev** [comment]
@Lineflyer @vladh 

I cannot replicate the same issues. Please post the raw signature you're using here so I can test locally.

Cheers.

**Comment by ntozier** [comment]
I am also unable to replicate this report.

@vladh what version are you running?

**Comment by protich** [comment]
Perhaps you edited the reply template and removed the signature variable?

**Comment by JediKev** [comment]
@Lineflyer @vladh 

You guys should go to **Admin Panel > Emails > Templates**, click the Template Set being used, and open the **Response/Reply Template**. Here you should see a signature variable as `%{signature}`:
![Screen Shot 2020-01-30 at 09 25 04](https://user-images.githubusercontent.com/11823401/73463346-b9a93c80-4342-11ea-8de6-1954e582e092.png)

If you do not see this variable then you will need to add it back, Save Changes, and retest.

Cheers.

**Comment by Lineflyer** [comment]
I did a quick check into the template config and can confirm the variable is missing.
Need to test though if its working after adding it, but I guess that was the problem here.
Thanks so much, I checked it several times but missed the missing ;)
